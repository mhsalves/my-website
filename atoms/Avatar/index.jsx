import React from 'react';
import { string, bool } from 'prop-types';

import {
  AvatarContainer,
  AvatarContent,
  AvatarImage,
} from './styles';

function Avatar(props) {
  const { src, alt, ...other } = props;

  return (
    <AvatarContainer {...other}>
      <AvatarContent {...other}>
        <AvatarImage src={src} alt={alt} />
      </AvatarContent>
    </AvatarContainer>
  );
}

Avatar.propTypes = {
  src: string,
  alt: string.isRequired,
  rounded: bool,
};

Avatar.defaultProps = {
  src: '',
  rounded: true,
};

export default Avatar;
