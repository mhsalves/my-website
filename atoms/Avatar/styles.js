import styled, { css } from 'styled-components';

const roundedStyle = css`
  border-radius: 50%;
  overflow: hidden;
`;

const AvatarContainer = styled.div`
  padding: ${({ theme: { space } }) => space('small')};
  background-color: rgba(255, 255, 255, .6);

  ${({ rounded }) => rounded && roundedStyle}
`;

const AvatarContent = styled.div`
  width: 100%;
  padding-top: 50%;
  padding-bottom: 50%;
  position: relative;

  ${({ rounded }) => rounded && roundedStyle}
`;

const AvatarImage = styled.img`
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;

  height: 100%;
`;

export {
  AvatarContainer,
  AvatarContent,
  AvatarImage,
};
