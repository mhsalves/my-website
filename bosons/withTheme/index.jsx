import React from 'react';
import { ThemeProvider } from 'styled-components';
import baseTheme from '../themes/base';

const withTheme = Component => props => (
  <ThemeProvider theme={baseTheme}>
    <Component {...props} />
  </ThemeProvider>
);

export default withTheme;
