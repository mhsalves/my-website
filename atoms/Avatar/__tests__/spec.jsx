import React from 'react';
import renderer from 'react-test-renderer';

import Avatar from '../index';

import withTheme from '../../../bosons/withTheme';

const ComponentWithTheme = withTheme(Avatar);

const src = 'https://via.placeholder.com/320';
const alt = 'Avatar testing';

describe('given: Avatar', () => {
  test('when: is not rounded', () => {
    const component = renderer.create(
      <ComponentWithTheme src={src} alt={alt} />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  test('when: is rounded', () => {
    const component = renderer.create(
      <ComponentWithTheme src={src} alt={alt} rounded />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
