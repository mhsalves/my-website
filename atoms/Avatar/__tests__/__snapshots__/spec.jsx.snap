// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given: Avatar when: is not rounded 1`] = `
.c0 {
  padding: 16px;
  background-color: rgba(255,255,255,.6);
}

.c1 {
  width: 100%;
  padding-top: 50%;
  padding-bottom: 50%;
  position: relative;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  height: 100%;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      alt="Avatar testing"
      className="c2"
      src="https://via.placeholder.com/320"
    />
  </div>
</div>
`;

exports[`given: Avatar when: is rounded 1`] = `
.c0 {
  padding: 16px;
  background-color: rgba(255,255,255,.6);
  border-radius: 50%;
  overflow: hidden;
}

.c1 {
  width: 100%;
  padding-top: 50%;
  padding-bottom: 50%;
  position: relative;
  border-radius: 50%;
  overflow: hidden;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  height: 100%;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      alt="Avatar testing"
      className="c2"
      src="https://via.placeholder.com/320"
    />
  </div>
</div>
`;
